### PR TITLE
Allow for overriding Default Providers on Installed components

### PR DESCRIPTION
**Pull request by web3devin**
* Add flag to Database Provider for overriding
* Add flag to Logging Provider for overriding
* Add flag to Queue Provider for overriding

These flags allow you to override the Default Providers with your own implementation. The flags default to `true` so that the `Laravel-Zero/Framework` package works right out of the box.

These changes are in reference to https://github.com/laravel-zero/laravel-zero/issues/203.

> These changes add flags to the Installed `Components'` `isAvailable()` method. This flag allows for a developer to easily override the default provider for that installed package.
> 
> One concern that exemplifies this is if you wanted to use the `Database Component`, but needed a way to override the `Migrator` for some reason. [With how `Service Providers` are loaded](https://github.com/laravel-zero/framework/blob/stable/src/Bootstrap/RegisterProviders.php#L69) the default providers are actually loaded in after the Custom Providers. This makes it very hard to override singletons that are built from the Default Providers.

### CODE REVIEW

**Comment by web3devin** [comment]
@nunomaduro  I opened this PR in a draft, as I wasn't sure if this was the method that you wanted to go with for extending add-ons. 

If any changes are needed, let me know as I have some time this week to work on this PR.

**Comment by pxgamer** [comment]
Sorry for the delay on this, I think this is fine to merge for the next release. 👍 Would be a great new feature. I do feel that something like how @nunomaduro had described in https://github.com/laravel-zero/laravel-zero/issues/203#issuecomment-484496602 would be slightly neater, but this is easier to implement and work with. 👍

File: src/Components/Queue/stubs/queue.php [hunk 1]
```diff
@@ -2,6 +2,19 @@
 
 return [
 
+    /*
```
Comment (nunomaduro) [inline src/Components/Queue/stubs/queue.php]: Can we remove this from the config, and document that people can actually override this here?

Comment (pxgamer) [reply src/Components/Queue/stubs/queue.php]: Remove it from the configuration stubs by default and just mention it in the docs? Or have I misunderstood?

Comment (nunomaduro) [reply src/Components/Queue/stubs/queue.php]: Yeah - just default this option to false on the internals of the framework, and if people want to change it they can do it here.

Comment (pxgamer) [reply src/Components/Queue/stubs/queue.php]: Sounds good, I'll open a PR soon. :+1:

**Pull request merged**